kinvolk/headlamp
# Reflect tables' page properties in the URL

These changes introduce a new `useURLState` which is the same as `React.useState` but reflects that state as a URL search parameter. For flexibility it can also be used as a plain `useState`, if the `key` passed to it is false.

It also makes sure that all tables in Headlamp are reflected in the URL.

The default values of the page and rowsPerPage are not shown in the URL (maybe for the latter it'd be a good idea since users may have a different default number for it but I haven't changed it).

Tests:
1. Go to a view that has a table; change the page and rows per page -> you should see those properties reflected in the URL.
2. Arrive at a list view with the p=2 (and enough data for those pages) -> you should be at page 2
3. Go to a URL with p=200000 -> it should show the last page
4. Go to a URL with p=-1 or perPage=-1 it should show the minimal value possible in each case

Try also the storybook with the new stories under SimpleTable.

fixes #21 

Needed for #781
